llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -1170,10 +1204,17 @@ class AllocaSlices::SliceBuilder : public PtrUseVisitor<SliceBuilder> {
     if (!IsOffsetKnown)
       return PI.setAborted(&II);
 
+    bool Splittable;
+
+    if (getVectorTypeFor(II, DL))
+      Splittable = isSplittableMemOp(AS.AI.getAllocatedType(), II.isVolatile());
```

Comment:
I don't think checking AI.getAllocatedType() here makes a lot of sense, seeing as how the memset may only be operating on part of it.